qa-tests-backend/scripts/run-compatibility.sh

```diff
@@ -20,6 +20,22 @@ source "$ROOT/qa-tests-backend/scripts/lib.sh"
 
 set -euo pipefail
 
+shorten_tag() {
+    if [[ "$#" -ne 1 ]]; then
+        die "Expected a version tag as parameter in shorten_tag: shorten_tag <tag>"
+    fi
+
+    long_tag=$1
+
+    short_tag_regex='([0-9]+\.[0-9]+\.[0-9xX]+)'
+
+    if [[ $long_tag =~ $short_tag_regex ]]; then
+        echo "${BASH_REMATCH[1]}"
+    else
+        die "Failed to shorten tag ${long_tag} as it did not match the regex: \"${short_tag_regex}\""
```

Comment:
You return an error if something cannot be shortened. However, further in this PR you do not check whether the error has occurred. How have you tested the `compatibility_test` when `shorten_tag` lands in this branch with `die`?

Reply:
Oh, good catch. As is, shorten_tag() would exit 1 and echo an error message that would then be parsed as the version and cause the test to fail later. I'll catch the exit 1 and output a proper error instead.

Reply:
Jan, I was thinking about something simpler - in case of an error, we do not want to kill the script. I would suggest to:

1) return the original value and print an error to stderr, but let the script continue
```suggestion
        echo "${long_tag}"
        >&2 echo "Failed to shorten tag ${long_tag} as it did not match the regex: \"${short_tag_regex}\""
```

2) Leave out the error checking in form:

```
if [ ! "$?" ]; then
   die ...
```

There is no point in calling `die` when you do the error handling outside of the function. The previous version with `die` inside `shorten_tag` did exactly the same.
The error is not fatal, so there is no need to immediately stop the script. Instead, we want to fallback to the long name and signal the error (see point 1). Do you think that makes sense?

Reply:
Oh that's how you meant it. Sure I can do that, it's a lot less code too. Although the behavior of the program would not be much different, since a tag is valid IFF it matches the tag regex.